### PR TITLE
EOS-23915: Bucket creation fails on online node in degraded cluster

### PR DESCRIPTION
**Pull request by imvenkip**
# Problem Statement
Bucket creation fails on online node in degraded cluster

# Design
This problem will solve with two fixes.
     1. PVER dirty flag validation (EOS-23915)
     2.  CAS request should succeed even one put request is success (EOS-27288)
  
This patch will solve PVER dirty flag issue.

Analysis:
Bucket creation is failed after one node down due to DIX pool dirty flag.

Fix:
Cluster have one DIX pool version, so removed dirty flag validation.
Enable this logic once multiple DIX pvers available in future.

# Coding
   Checklist for Author
-  [X] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [X] JIRA number/GitHub Issue added to PR
- [X] PR is self reviewed
- [X] Jira and state/status is updated and JIRA is updated with PR link
- [X] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: motr/setup_dix.c [hunk 1]
```diff
@@ -344,7 +344,10 @@ M0_INTERNAL int m0_cs_dix_setup(struct m0_motr *cctx)
 	M0_ASSERT(pver != NULL);
 	pm = &pver->pv_mach;
 	m0_rwlock_read_lock(&pm->pm_lock);
-	M0_ASSERT(!pver->pv_is_dirty);
+	/*
+ 	 * Enable this logic once multiple DIX pvers available.
```
Comment (yatin-mahajan) [inline motr/setup_dix.c:348]: [Suggestion]
It will be good to add FIXME/TODO kind of label here as this is AI for future.
same for below comment.

**Review by yatin-mahajan — APPROVED** [review]
Looks good to me.

File: motr/setup_dix.c (hunk 1 above)
Comment (imvenkip) [reply motr/setup_dix.c]: Done.

File: motr/setup_dix.c [hunk 2]
```diff
@@ -344,7 +344,10 @@ M0_INTERNAL int m0_cs_dix_setup(struct m0_motr *cctx)
 	M0_ASSERT(pver != NULL);
 	pm = &pver->pv_mach;
 	m0_rwlock_read_lock(&pm->pm_lock);
-	M0_ASSERT(!pver->pv_is_dirty);
+	/*
+ 	 * @todo: Enable this logic once multiple DIX pvers available. 
+         */ 
+	// M0_ASSERT(!pver->pv_is_dirty);
```
Comment (madhavemuri) [inline motr/setup_dix.c:350]: Move this line to above commented code.
In motr C++ style commenting is not used.

Comment (imvenkip) [reply motr/setup_dix.c]: Done

**Review by madhavemuri — APPROVED** [review]

**Comment by mehjoshi** [comment]
retest this please

**Review by mehjoshi — APPROVED** [review]
LGTM.

**Pull request merged**